docs/CppInteroperability/Roadmap.md

```diff
@@ -0,0 +1,141 @@
+
+For example, “virtual type interfaces that API consumers provide implementations for” are a fairly common API pattern in C++. Currently, there is no design for such a pattern, but some Swift contributor may extend interop to provide a clear, native mapping for this API pattern through a self-contained proposal following the goals outlined by this document, and framed by “the approach” described above. 
+
+## The Swift ecosystem
+
+### Tooling and build process
+
+It goes without saying (yet will be said anyway) that as a supported language feature, C++ and Swift interoperability must work well on every platform supported by Swift. In a similar vein, tools in the Swift ecosystem should be updated to support interoperability features. For example, SourceKit should provide autocompletion, jump-to-definition, etc. for C++ functions, methods, and types and lldb should be able to print C++ types (even in Swift frames). Finally, the Swift package manager should be updated with the necessary features to support building C++ dependencies, a topic which Saleem Abdulrasool may be able to expand on here :)
+
+This roadmap outlines a strategy for importing APIs that relies on semantic information from the user. In order to make this painless for users across a variety of projects, Swift will need to provide both inline annotation support for C++ APIs and side-file support for APIs that cannot be updated. For Objective-C, this side-file is an APINotes file. As part of Swift and C++ interoperability, APINotes will either need to be updated to support C++ APIs, or another kind side-file will need to be created. 
+
+### The standard library
+
+Egor Zhdan will discuss importing the standard library, the Swift C++ standard library overlay, etc. :)
+* * *
+
+## Thoughts not incorporated into the above document:
+
+* Maybe discuss drawing lines for other things. The patterns you cover talk about memory ownership specifically. What about virtual inheritance? What about templates? What about mutexes and file opens?
+* Safety: It is convention in C and Objective-C to traffic in pointers. In Swift this is written as “unsafe” meaning the unsafety is clearly disclosed. C++ often prefers value types and “fancy pointers” which do not disclosure their unsafety in their name.
+* Safety: because C++‘s un-oppionated APIs are funneled into Swift, they are generally forced through unsafe pointer which loses information. This loss of semantic information leads to mis-use, if more information was captured it would be more difficult, or even impossible, to make such mistakes. 
```

Comment:
```suggestion
* Safety: because C++‘s un-opinionated APIs are funneled into Swift, they are generally forced through unsafe pointer which loses information. This loss of semantic information leads to misuse, if more information was captured it would be more difficult, or even impossible, to make such mistakes. 
```